core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java

```diff
@@ -154,52 +155,68 @@ public boolean delete() {
+      DataOutputStream dataOutputStream = null;
+      DataInputStream dataInputStream = null;
+      // if bytes to read less than 1024 then buffer size should be equal to the given offset
+      int bufferSize = validDataEndOffset > CarbonCommonConstants.BYTE_TO_KB_CONVERSION_FACTOR ?
+          CarbonCommonConstants.BYTE_TO_KB_CONVERSION_FACTOR :
+          (int) validDataEndOffset;
+      // temporary file name
+      String tempWriteFilePath = fileName + CarbonCommonConstants.TEMPWRITEFILEEXTENSION;
+      FileFactory.FileType fileType = FileFactory.getFileType(fileName);
+      try {
+        CarbonFile tempFile;
+        // delete temporary file if it already exists at a given path
+        if (FileFactory.isFileExist(tempWriteFilePath, fileType)) {
+          tempFile = FileFactory.getCarbonFile(tempWriteFilePath, fileType);
+          tempFile.delete();
         }
-        read = dataInputStream.read(buff, 0, buff.length);
+        // create new temporary file
+        FileFactory.createNewFile(tempWriteFilePath, fileType);
+        tempFile = FileFactory.getCarbonFile(tempWriteFilePath, fileType);
+        byte[] buff = new byte[bufferSize];
+        dataInputStream = FileFactory.getDataInputStream(fileName, fileType);
+        // read the data
+        int read = dataInputStream.read(buff, 0, buff.length);
+        dataOutputStream = FileFactory.getDataOutputStream(tempWriteFilePath, fileType);
         dataOutputStream.write(buff, 0, read);
-        remaining = remaining - read;
+        long remaining = validDataEndOffset - read;
+        // anytime we should not cross the offset to be read
+        while (remaining > 0) {
+          if (remaining > bufferSize) {
+            buff = new byte[bufferSize];
+          } else {
+            buff = new byte[(int) remaining];
+          }
+          read = dataInputStream.read(buff, 0, buff.length);
+          dataOutputStream.write(buff, 0, read);
+          remaining = remaining - read;
+        }
+        CarbonUtil.closeStreams(dataInputStream, dataOutputStream);
+        // rename the temp file to original file
+        tempFile.renameForce(fileName);
+        fileTruncatedSuccessfully = true;
+      } catch (IOException ioe) {
+        LOGGER.error("IOException occurred while truncating the file " + ioe.getMessage());
+      } finally {
+        CarbonUtil.closeStreams(dataOutputStream, dataInputStream);
       }
-      CarbonUtil.closeStreams(dataInputStream, dataOutputStream);
-      // rename the temp file to original file
-      tempFile.renameForce(fileName);
-      fileTruncatedSuccessfully = true;
     } catch (IOException e) {
-      LOGGER.error("Exception occurred while truncating the file " + e.getMessage());
-    } finally {
-      CarbonUtil.closeStreams(dataOutputStream, dataInputStream);
+      LOGGER.error("IOException occurred while truncating the file " + e.getMessage());
+    } catch (Exception e) {
```

Comment:
why finally, need add the catch again : 

catch (Exception e) {
	  LOGGER.error("Other exception occurred while truncating the file + e.getMessage());
      }

Reply:
Method.invoke may throw other exceptions when running, such as: IllegalAccessException, 
 IllegalArgumentException, InvocationTargetException;
So need to add this catch for above exceptions.

The finally block is for the try block which is in catch NoSuchMethodException branch.